iree/compiler/Codegen/LLVMCPU/test/transpose_avx2_lowering.mlir

```diff
@@ -0,0 +1,119 @@
+// RUN: iree-opt -pass-pipeline='hal.executable(hal.executable.variant(iree-llvmcpu-lower-executable-target{test-lowering-configuration=false}))' -split-input-file %s | FileCheck %s
+
+#executable_layout = #hal.executable.layout<push_constants = 0, sets = [
+  #hal.descriptor_set.layout<0, bindings = [
+    #hal.descriptor_set.binding<0, storage_buffer>,
+    #hal.descriptor_set.binding<1, storage_buffer>
+  ]>
+]>
+#executable_target_embedded_elf_x86_64_ = #hal.executable.target<
+  "llvm",
+  "embedded-elf-x86_64", {
+    cpu_features = "+avx2",
+    data_layout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128",
+    native_vector_size = 32 : index,
+    target_triple = "x86_64-unknown-unknown-eabi-elf"
+  }>
+
+
+hal.executable private @test_8x8_avx2_pattern {
+  hal.executable.variant @embedded_elf_x86_64, target = #executable_target_embedded_elf_x86_64_ {
+    hal.executable.entry_point @test_8x8_avx2_pattern layout(#executable_layout)
+    builtin.module {
+      func @test_8x8_avx2_pattern() {
+        %cst = arith.constant 0.000000e+00 : f32
+        %c0 = arith.constant 0 : index
+        %0 = hal.interface.binding.subspan set(0) binding(0) type(storage_buffer) offset(%c0) alignment(64) : !flow.dispatch.tensor<readonly:512x1024xf32>
+        %2 = hal.interface.binding.subspan set(0) binding(1) type(storage_buffer) offset(%c0) alignment(64) : !flow.dispatch.tensor<writeonly:1024x512xf32>
+        %3 = flow.dispatch.tensor.load %0, offsets = [0, 0], sizes = [512, 1024], strides = [1, 1] : !flow.dispatch.tensor<readonly:512x1024xf32> -> tensor<512x1024xf32>
+        %5 = flow.dispatch.tensor.load %2, offsets = [0, 0], sizes = [1024, 512], strides = [1, 1] : !flow.dispatch.tensor<writeonly:1024x512xf32> -> tensor<1024x512xf32>
+        %6 = linalg.generic {
```

Comment:
Does this require the op to be a "transpose" only or any generic op would work?

Reply:
The current changes are not introducing anything at Linalg level, mostly because we got the expected tiling (8x8) for the AVX2 pattern to trigger, at least for this case. So these changes are only enabling the lowering of an 8x8 or 4x8 vector.transpose to AVX2. If that specific op is not found, nothing happens. Follow up PRs will "force" the 8x8 tiling (or 4x8) for more transpose cases so that we take advantage of this pattern more broadly.